packages/core/src/browser/cookie.ts

```diff
@@ -9,14 +11,18 @@ export interface CookieOptions {
   domain?: string
 }
 
-export function setCookie(name: string, value: string, expireDelay: number, options?: CookieOptions) {
+export function setCookie(name: string, value: string, expireDelay: number = 0, options?: CookieOptions) {
   const date = new Date()
   date.setTime(date.getTime() + expireDelay)
-  const expires = `expires=${date.toUTCString()}`
+  let expires = `expires=${date.toUTCString()}`
   const sameSite = options && options.crossSite ? 'none' : 'strict'
   const domain = options && options.domain ? `;domain=${options.domain}` : ''
   const secure = options && options.secure ? ';secure' : ''
   const partitioned = options && options.partitioned ? ';partitioned' : ''
+  // we cannot read from cookie here because it will affect cookie locking
+  if (isExperimentalFeatureEnabled(ExperimentalFeature.ANONYMOUS_USER_TRACKING) && name === SESSION_STORE_KEY) {
+    expires = `max-age=${ONE_YEAR}`
+  }
```

Comment:
**💬 suggestion:** you could just provide a bigger `expireDelay` argument when setting the session cookie instead of having this special case